### DISCUSSION TITLE
zh/config/configuration

### DISCUSSION THREAD

**Discussion by giscus** [AlistGo/docs#36] [Announcements]
# zh/config/configuration

AList v3的文档

https://alist.nn.ci/zh/config/configuration.html

**Comment by jiyexing** [comment]
我想固定地址
我将"address": "0.0.0.0",
改为"address": "192.168.0.43",
无法开启服务

**Reply by jiyexing** [reply]
请问我该如何做（win设备）

**Reply by jiyexing** [reply]
受教了

**Comment by aoxiangtianji** [comment]
database 中的 ssl_mode 值为 disable / enable
踩坑记录

**Reply by yh6441** [reply]
请教如何改呢？

**Comment by leiyuanxi** [comment]
请问
site_url字段需要填写端口吗？

**Reply by fisschl** [reply]
需要

**Comment by X-yael** [comment]
感觉像是开始了防盗链一样，无法引用cdn1.tianli0.top的前端文件，包括演示站点也一样，打开网站会一片空白，改成jsdelivr就没问题了

**Reply by Daxhu18** [reply]
兄弟怎么改啊？

**Reply by Toserk1024** [reply]
应该是跨域问题

**Reply by LeAnHe1713** [reply]
在config.json文件里把cdn改成jsdelivr的了，还是一片空白。挂梯子后，手机用edge浏览器也只能用桌面模式才能显示出内容。。

**Reply by LeAnHe1713** [reply]
另外，用联通卡的数据网络就没这个问题。电信的数据网络会有这个问题。

**Comment by zxcv-12345** [comment]
配置文件路径：/opt/alist/data/config.json

**Reply by leosy2333** [reply]
正好找你这条评论了 顶一下

**Comment by xzxiaoshan** [comment]
临时目录中的文件，在使用后不会被自动清理，且界面上没有清理缓存的地方。
长期以往，累积缓存文件占用磁盘。

**Comment by wotime** [comment]
静态资源CDN增添补充一个：https://cdn.onmicrosoft.cn/，实测体验还不错，可以增加为备用。
![1673269495328.png](https://bj.bcebos.com/baidu-rmb-video-cover-1/14308e61f1ef90d018b9e73e3218d895.png)

**Reply by Kvinam** [reply]
怎么使用

**Comment by Aziily** [comment]
请问site_url应该如何写，我这样写是不可以吗
"site_url": "http://localhost:5244/alist"
可以的话能不能给个例子参考一下（

**Reply by Aziily** [reply]
目的是为了配置群晖nas的反向代理，然后希望能在上述地址正常访问alist

**Comment by cancundeyingzi** [comment]
新版好难用,比如日志......2那多方便,3连个记录详细程度都没

**Comment by cancundeyingzi** [comment]
site_url不写,视频放不出来,写了,,网页都进不去....服了

**Reply by cancundeyingzi** [reply]
我还有不同的ip地址.....

**Reply by cancundeyingzi** [reply]
已解决

**Reply by snailwind** [reply]
请问如何配置的？site_url不知道怎么写，反带只要不是根目录就网页白页

**Comment by kimycai** [comment]
如果是windows系统，那么SSL证书存放路径格式应该怎么写？和linux相同吗，还是data/public.crt？

**Comment by cancundeyingzi** [comment]
token_expires_in9999999999无效

**Comment by wantong-lab** [comment]
配置https证书之后，存储配置的302重定向，无法下载文件了。

**Reply by imysl0** [reply]
http 和 https 的端口貌似无法设置成同一个端口

**Comment by zzerosama** [comment]
无证书，可以开启https吗？

**Comment by liwezee** [comment]
我想问一下，就是我的网站的域名是https://liwezee.site，我在网站上的服务器上部署了alist，在配置文件里设置进alist的域名为https://cloud.liwezee.site，https的证书也放在了data里，而且也将dns解析到了服务器，但是我访问 https://cloud.liwezee.site 的时候却不是alist的页面，而是我网站的页面 ，而且浏览器显示网站不安全，提示证书无效，我试了很久都不行，而且我用服务器ip+端口是能进去的，但是就是无法用 “https://cloud.liwezee.site”这个域名访问，用这个域名访问就会直接显示我网站的主页，我的网站和alist是同一个服务器的，而且服务器是有安装宝塔面板的，求大佬解答

**Comment by mapleshadow** [comment]
"https_port": -1,设置为"https_port": 5245,之后，报错，提示：failed to start https: open : no such file or directory
还请修复。这个是不是在提示没找到TLS证书？？？

**Reply by mapleshadow** [reply]
已解决，果然是要把网站TLS证书放入，不然就会报错

**Reply by Turkeyer** [reply]
你好，我看到已解决了，想请教一下，我已经把xx.cer和xx.key放在了alist的config.json同目录下。修改如下 
"scheme": {
    "address": "0.0.0.0",
    "http_port": 5244,
    "https_port": 5245,
    "force_https": false,
    "cert_file": "xx.cer",
    "key_file": "xx.key",
    "unix_file": "",
    "unix_file_perm": ""
但alist启动不来，报错failed to start https: open xx.cer: no such file or directory 
请问你是怎么做的？谢谢

**Comment by yu4n2h1** [comment]
能配置ipv6吗,怎么改config才能使得ipv6生效

**Comment by John-Wong** [comment]
关于数据库设置的ssl_mode，查看GORM及数据库驱动文档得知：
- 对于MySQL，有：true, false, skip-verify, preferred，与文中所述有出入
> tls=true enables TLS / SSL encrypted connection to the server. Use skip-verify if you want to use a self-signed or invalid certificate (server side) or use preferred to use TLS only when advertised by the server. This is similar to skip-verify, but additionally allows a fallback to a connection which is not encrypted. Neither skip-verify nor preferred add any reliable security. You can use a custom TLS config after registering it with mysql.RegisterTLSConfig.
- 对于postgres，有：disable, allow, prefer, require, verify-ca, verify-full，与文中所述相同
